repository.go

```diff
@@ -563,7 +563,14 @@ func (r *Repository) SetConfig(cfg *config.Config) error {
 // are returned merged in one config value.
 func (r *Repository) ConfigScoped(scope config.Scope) (*config.Config, error) {
 	// TODO(mcuadros): v6, add this as ConfigOptions.Scoped
+	if scope.IsV6PreviewScope() {
+		return r.configScopedV6(scope)
+	}
+
+	return r.configScopedV5(scope)
```

Comment:
I would bring the `configScopedV5` and `configScopedV6` logic into `config/config.go` and have a simple call to it here. Potentially following the same approach we discussed above:


```suggestion
	return scope.ConfigScoped()
```

Reply:
This refactor may be trickier than the others, as it hits on one of the complications with the current approach.

The `Repository` has a `Storer` that acts as an abstraction over repository storage; this embeds a `config.ConfigStorer` which seems to be used to keep a copy of local config in memory.

git config commands generally act on the config in one of the stored locations (local, global, system, worktree, or file) and then complete with the config flushed to disk (or its `Storer` equivalent). subsequent commands then act on the resulting merged config by default unless one of those scope arguments is added.

In other words, go-git v5 acts on local configuration alone by default, while git config acts on a merged system/global/local config by default. This is the root cause of issues like #269 and #239 and impacts issues like #830

I will have to reflect on the implications of how to push this Repository-scoped function into the Config package.

Reply:
Turns out that this is easier than I thought as the existing interface structure supports this refactor quite nicely.

Because `storage.Storer` embeds `config.ConfigStorer` and these methods depend only on the `config.ConfigStorer` part of `storage.Storer` these methods can move into the `config` package quite easily.